`mr"“`/`mr"”` replaces double quotes with two pairs of the same kind of double quote

### Summary

I added the following to my helix config:

```toml
[editor.auto-pairs]
'(' = ')'
'{' = '}'
'[' = ']'
'"' = '"'
"'" = "'"
'“' = '”'
# ’ is usually an apostrophe
'«' = '»'
'`' = '`'
```

and tried to replace straight double quotes with curly double quotes. However, instead of getting leading and trailing double quotes, I got two of the same kind of whatever I typed in.

I tried this with `mr"«` and `mr"»` and it didn't screw up like this. It worked right for the guillemets.

### Reproduction Steps

<!-- Ideally provide a key sequence and/or asciinema.org recording. --> 

I tried this:

1. run `hx`.
2. insert `"Do they allow sheep on the airplane?"` into the buffer however you like.
3. Get into normal mode by pressing escape.
4. type `mr"“` (`⎄"<` works, where ⎄ is your Compose key, or type `⌥[` in macOS)

I expected this to happen:

Both the leading and trailing double quotes should be curled differently from each other.

Instead, this happened:

Both the leading and trailing double quotes are curled the same way (in this case, leading).

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
2024-03-22T23:26:40.960 helix_view::editor [ERROR] Failed to initialize the language servers for `source.md` - `marksman` { cannot find binary path }
2024-04-19T00:20:24.456 helix_core::syntax [ERROR] Failed to load tree-sitter parser for language "gemini": Error opening dynamic library "/home/linuxbrew/.linuxbrew/Cellar/helix/24.03/bin/runtime/grammars/gemini.so"
2024-06-14T23:22:03.428 helix_core::syntax [ERROR] Failed to load tree-sitter parser for language "gemini": Error opening dynamic library "/home/linuxbrew/.linuxbrew/Cellar/helix/24.03/bin/runtime/grammars/gemini.so"

```

</details>


### Platform

macOS and Linux (Debian Stable)

### Terminal Emulator

Terminal.app, iTerm2.app, gnome-terminal

### Installation Method

Homebrew (on both macOS _and_ Debian)

### Helix Version

24.3 (Homebrew says `24.03` in both cases, though)

--- a/helix-core/src/match_brackets.rs
+++ b/helix-core/src/match_brackets.rs
@@ -9,11 +9,13 @@ use crate::Syntax;
 const MAX_PLAINTEXT_SCAN: usize = 10000;
 const MATCH_LIMIT: usize = 16;
 
-pub const BRACKETS: [(char, char); 7] = [
+pub const BRACKETS: [(char, char); 9] = [
     ('(', ')'),
     ('{', '}'),
     ('[', ']'),
     ('<', '>'),
+    ('‘', '’'),
+    ('“', '”'),
     ('«', '»'),
     ('「', '」'),
     ('（', '）'),

